test/multiverse/suites/elasticsearch/elasticsearch_instrumentation_test.rb

```diff
@@ -0,0 +1,175 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require 'elasticsearch'
+require 'socket'
+
+class ElasticsearchInstrumentationTest < Minitest::Test
+  def setup
+    # Keeping the log off prevents noisy test output
+    @client = ::Elasticsearch::Client.new(
+      log: false,
+      hosts: "localhost:#{port}"
+    )
+    # Ensure the client is running before the tests start
+    @client.cluster.health
+    @client.index(index: 'my-index', id: 1, body: {title: 'Test'})
+    @client.indices.refresh(index: 'my-index')
+  end
+
+  def teardown
+    @client = nil
+    @segment = nil
+  end
+
+  def search
+    txn = in_transaction do
+      @client.search(index: 'my-index', body: {query: {match: {title: 'test'}}})
+    end
+    @segment = txn.segments[1]
+  end
+
+  # TODO! TEST METRIC GENERATION!!!
```

Comment:
I think this TODO has either been satisfied or can be tracked elsewhere outside of the code now.